ark/internal/controller/tool_controller.go

```diff
@@ -47,5 +52,92 @@ func (r *ToolReconciler) updateToolStatus(ctx context.Context, tool *arkv1alpha1
 }
 
 func (r *ToolReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	if err := mgr.Add(&builtinToolInitializer{client: mgr.GetClient()}); err != nil {
+		return fmt.Errorf("failed to add builtin tool initializer: %w", err)
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).For(&arkv1alpha1.Tool{}).Named("tool").Complete(r)
 }
+
+type builtinToolInitializer struct {
+	client client.Client
+}
+
+func (b *builtinToolInitializer) Start(ctx context.Context) error {
+	log := logf.FromContext(ctx)
+	log.Info("initializing built-in tools")
+
+	if err := b.ensureBuiltinToolsExist(ctx); err != nil {
+		log.Error(err, "failed to create built-in tools")
+		return fmt.Errorf("failed to create built-in tools: %w", err)
+	}
+
+	log.Info("built-in tools initialization completed")
+	return nil
+}
+
+func (b *builtinToolInitializer) ensureBuiltinToolsExist(ctx context.Context) error {
+	builtinTools := []genai.ToolDefinition{
+		genai.GetNoopTool(),
+		genai.GetTerminateTool(),
+	}
+
+	for _, builtinTool := range builtinTools {
+		if err := b.ensureSingleBuiltinTool(ctx, builtinTool.Name, builtinTool.Description, builtinTool.Parameters); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (b *builtinToolInitializer) ensureSingleBuiltinTool(ctx context.Context, name, description string, parameters map[string]any) error {
+	log := logf.FromContext(ctx)
+	log.Info("ensuring built-in tool exists", "tool", name)
```

Comment:
for these built in tools it can probably be debug logging and only when something changes rather than in all cases (e.g. we debug log if we need to add it)